File: 20180622_SDK/SDK/Sample_M50/C#_SBXPCDLL_Sample/SBXPCDLLSampleCSharp/frmDepartment.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;

using System.Runtime.InteropServices;

namespace SBXPCDLLSampleCSharp
{
    public partial class frmDepartment : Form
    {
        public frmDepartment()
        {
            InitializeComponent();
        }

        const int DEPT_COUNT = 20;
        string[] departmentNameList;

        private void frmDepartment_Load(object sender, EventArgs e)
        {
            departmentNameList = new string[DEPT_COUNT];
            DrawDepartmentList();
        }

        private void DrawDepartmentList()
        {
            lstDepartment.Items.Clear();
            string itemString;
            for (int i = 0; i < DEPT_COUNT; i++)
            {
                itemString = "[No.]" + String.Format("{0:D2}", i) + " ";
                itemString += "[Name]" + departmentNameList[i];
                lstDepartment.Items.Add(itemString);
            }
        }

        private void lstDepartment_SelectedIndexChanged(object sender, EventArgs e)
        {
            txtDeparmtmentName.Text = departmentNameList[lstDepartment.SelectedIndex];
            txtDeparmtmentName.Focus();
        }

        private void cmdUpdate_Click(object sender, EventArgs e)
        {
            if (lstDepartment.SelectedIndex == -1) return;
            departmentNameList[lstDepartment.SelectedIndex] = txtDeparmtmentName.Text;
            DrawDepartmentList();
        }

        private void frmDepartment_FormClosing(object sender, FormClosingEventArgs e)
        {
            Application.OpenForms["frmMain"].Visible = true;
        }

        private void cmdExit_Click(object sender, EventArgs e)
        {
            Close();
        }

        private void cmdRead_Click(object sender, EventArgs e)
        {
            bool bRet;
            int vErrorCode = 0;

            lblMessage.Text = "Working...";
            Application.DoEvents();

            bRet = sbxpc.SBXPCDLL.EnableDevice(Program.gMachineNumber, 0);
            if(!bRet)
            {
                lblMessage.Text = util.gstrNoDevice;
                return;
            }

            for(int i = 0; i < DEPT_COUNT; i ++)
                bRet = sbxpc.SBXPCDLL.GetDepartName(Program.gMachineNumber, i, 0, out departmentNameList[i]) && bRet;
            if(bRet)
                lblMessage.Text = "Success";
            else
            {
                sbxpc.SBXPCDLL.GetLastError(Program.gMachineNumber, out vErrorCode);
                lblMessage.Text = util.ErrorPrint(vErrorCode);
            }

            bRet = sbxpc.SBXPCDLL.EnableDevice(Program.gMachineNumber, 1);

            DrawDepartmentList();
        }

        private void cmdWrite_Click(object sender, EventArgs e)
        {
            bool bRet;
            int vErrorCode = 0;

            lblMessage.Text = "Working...";
            Application.DoEvents();

            bRet = sbxpc.SBXPCDLL.EnableDevice(Program.gMachineNumber, 0);
            if (!bRet)
            {
                lblMessage.Text = util.gstrNoDevice;
                return;
            }

            for (int i = 0; i < DEPT_COUNT; i++)
                bRet = sbxpc.SBXPCDLL.SetDepartName(Program.gMachineNumber, i, 0, departmentNameList[i]) && bRet;
            if (bRet)
                lblMessage.Text = "Success";
            else
            {
                sbxpc.SBXPCDLL.GetLastError(Program.gMachineNumber, out vErrorCode);
                lblMessage.Text = util.ErrorPrint(vErrorCode);
            }

            bRet = sbxpc.SBXPCDLL.EnableDevice(Program.gMachineNumber, 1);

            DrawDepartmentList();
        }
    }
}

File: 20180622_SDK/SDK/Sample_M50/VB.NET_SBXBPCSample/frmDepartment.vb
Option Strict Off
Option Explicit On
Imports System.Runtime.InteropServices

Public Class frmDepartment
    Const DEPT_COUNT As Integer = 20
    Dim departmentNameList(DEPT_COUNT - 1) As String
    Private Sub frmDepartment_Load(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Load
        DepartmentNameInit()
        DrawDepartmentNameList()
    End Sub

    Private Sub DepartmentNameInit()
        Dim i As Integer
        For i = 0 To DEPT_COUNT - 1
            departmentNameList(i) = ""
        Next
    End Sub

    Private Sub DrawDepartmentNameList()
        Dim i As Integer
        lstDepartment.Items.Clear()
        For i = 0 To DEPT_COUNT - 1
            lstDepartment.Items.Add("[No.] " & VB6.Format(CStr(i), "0# ") & _
                                "[Name] " & departmentNameList(i))
        Next
    End Sub

    Private Sub cmdUpdate_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles cmdUpdate.Click
        If lstDepartment.SelectedIndex = -1 Then
            Exit Sub
        End If

        departmentNameList(lstDepartment.SelectedIndex) = txtDepartmentName.Text
        DrawDepartmentNameList()
    End Sub

    Private Sub lstDepartment_SelectedIndexChanged(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles lstDepartment.SelectedIndexChanged
        If lstDepartment.SelectedIndex = -1 Then
            Exit Sub
        End If

        txtDepartmentName.Text = departmentNameList(lstDepartment.SelectedIndex)
        txtDepartmentName.Focus()
    End Sub

    Private Sub frmDepartment_FormClosing(ByVal sender As System.Object, ByVal e As System.Windows.Forms.FormClosingEventArgs) Handles MyBase.FormClosing
        frmMain.Visible = True
    End Sub

    Private Sub cmdExit_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles cmdExit.Click
        Close()
    End Sub

    Private Sub cmdRead_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles cmdRead.Click
        Dim bRet As Boolean
        Dim vErrorCode As Integer
        Dim i As Integer
        Dim departmentName As String = ""

        lblMessage.Text = "Working..."
        Application.DoEvents()

        bRet = sbxpc.SBXPCDLL.EnableDevice(frmMain.gMachineNumber, False)
        If Not bRet Then
            lblMessage.Text = gstrNoDevice
            Exit Sub
        End If

        For i = 0 To DEPT_COUNT - 1

            bRet = sbxpc.SBXPCDLL.GetDepartName(frmMain.gMachineNumber, i, 0, departmentName)

            If bRet Then
                lblMessage.Text = "Success"
                departmentNameList(i) = departmentName
            Else
                sbxpc.SBXPCDLL.GetLastError(frmMain.gMachineNumber, vErrorCode)
                lblMessage.Text = ErrorPrint(vErrorCode)
                departmentNameList(i) = ""
            End If
        Next

        bRet = sbxpc.SBXPCDLL.EnableDevice(frmMain.gMachineNumber, True)

        DrawDepartmentNameList()
    End Sub

    Private Sub cmdWrite_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles cmdWrite.Click
        Dim bRet As Boolean
        Dim vErrorCode As Integer
        Dim i As Integer

        lblMessage.Text = "Working..."
        Application.DoEvents()

        bRet = sbxpc.SBXPCDLL.EnableDevice(frmMain.gMachineNumber, False)
        If Not bRet Then
            lblMessage.Text = gstrNoDevice
            Exit Sub
        End If

        For i = 0 To DEPT_COUNT - 1

            bRet = sbxpc.SBXPCDLL.SetDepartName(frmMain.gMachineNumber, i, 0, departmentNameList(i))

            If bRet Then
                lblMessage.Text = "Success"
            Else
                sbxpc.SBXPCDLL.GetLastError(frmMain.gMachineNumber, vErrorCode)
                lblMessage.Text = ErrorPrint(vErrorCode)
            End If
        Next

        bRet = sbxpc.SBXPCDLL.EnableDevice(frmMain.gMachineNumber, True)
    End Sub
End Class